src/views/Graph.vue

```diff
@@ -387,6 +542,209 @@ export default {
+        user: undefined,
+        workflow: undefined,
+        workflowID: ''
+      }
+      const newEdge = {
+        children: [],
+        familyTree: undefined,
+        id: edgeId,
+        name: edgeId,
+        node: {
+          id: edgeId,
+          source: `${edgePath}//${sourceId}`,
+          target: `${edgePath}//${targetId}`,
+        },
+        tokens: {
+          cycle: undefined,
+          edge: [sourceObject, targetObject],
+          id: edgeId,
+          job: undefined,
+          namespce: undefined,
+          relativeID: '',
+          task: undefined,
+          user: undefined,
+          workflow: undefined,
+          workflowID: edgePath
+        },
+        type: 'family'
+      }
+      return newEdge
+    },
+    removeEdge (sourceName, sourceCyclePoint, targetName, targetCyclePoint, edges) {
+      // removes a edge object from the 'edges' array
+      const edgePath = this.workflowIDs[0]
+      const edgeSearchTerm = `${edgePath}//$edge|${sourceCyclePoint}/${sourceName}|${targetCyclePoint}/${targetName}`
+      const edgesFiltered = edges.filter((edge) => {
+        if (edge.id.indexOf(edgeSearchTerm) === -1) {
+          return true
+        } else {
+          return false
+        }
+      })
+      return edgesFiltered
+    },
+    checkForEdgeBySource (sourceName, cyclePoint, edges) {
+      const edgePath = this.workflowIDs[0]
+      const edgeSearchTerm = `${edgePath}//${cyclePoint}/${sourceName}`
+
+      const edgeSearch = edges.filter((edge) => {
+        return edge.node.source === edgeSearchTerm
+      })
+      return edgeSearch
+    },
+    checkForEdgeByTarget (sourceName, cyclePoint, edges) {
+      const edgePath = this.workflowIDs[0]
+      const edgeSearchTerm = `${edgePath}//${cyclePoint}/${sourceName}`
+
+      const edgeSearch = edges.filter((edge) => {
+        return edge.node.target === edgeSearchTerm
+      })
+      return edgeSearch
+    },
+    removeEdgeBySource (edgeCheckSource, edges, removedEdges, config, cycle) {
+      edgeCheckSource.forEach((edge) => {
+        removedEdges.push(edge)
+        const targetName = this.cylcTree.$index[edge.node.target].name
+        const targetCycle = this.cylcTree.$index[edge.node.target].tokens.cycle
+        edges = this.removeEdge(config.name, cycle, targetName, targetCycle, edges)
+      })
+      return [edges, removedEdges]
+    },
+    removeEdgeByTarget (edgeCheckTarget, edges, removedEdges) {
+      edgeCheckTarget.forEach((edge) => {
+        removedEdges.push(edge)
+        const targetName = this.cylcTree.$index[edge.node.target].name
+        const targetCycle = this.cylcTree.$index[edge.node.target].tokens.cycle
+        const sourceName = this.cylcTree.$index[edge.node.source].name
+        const sourceCycle = this.cylcTree.$index[edge.node.source].tokens.cycle
+        edges = this.removeEdge(sourceName, sourceCycle, targetName, targetCycle, edges)
+      })
+      return [edges, removedEdges]
+    },
```

Comment:
The source and target are available in the tokens (`edge[0] == source`, `edge[1] == target`) which avoids the need to go via the `cylcTree.$index`:

```suggestion
    removeEdgeBySource (edgeCheckSource, edges, removedEdges, config, cycle) {
      edgeCheckSource.forEach((edge) => {
        removedEdges.push(edge)
        edges = this.removeEdge(
          config.name,
          cycle,
          edge.tokens.edge[1].task,
          edge.tokens.edge[1].cycle,
          edges
        )
      })
      return [edges, removedEdges]
    },
    removeEdgeByTarget (edgeCheckTarget, edges, removedEdges) {
      edgeCheckTarget.forEach((edge) => {
        removedEdges.push(edge)
        edges = this.removeEdge(
          edge.tokens.edge[0].task,
          edge.tokens.edge[0].cycle,
          edge.tokens.edge[1].task,
          edge.tokens.edge[1].cycle, edges
        )
      })
      return [edges, removedEdges]
    },
```